mesonbuild/ast/interpreter.py

```diff
@@ -235,7 +235,7 @@ def resolve_key(node: mparser.BaseNode) -> str:
         arguments, kwargs = self.reduce_arguments(node.args, key_resolver=resolve_key)
         assert not arguments
         self.argument_depth += 1
-        for key in kwargs.keys():
+        for key in kwargs:
```

Comment:
This seems to partially revert a change in the previous commit, so logically the first commit should be fixed instead.

Reply:
Let's unify the commits because they could all be made by a single ruff command.

This fix could have been automatically made by adding [`ruff rule SIM018`](https://docs.astral.sh/ruff/rules/in-dict-keys) to the `--select` statement.  However this would have changed another 23 instances across the codebase so it was not chosen to make this PR easier to review and merge.

Reply:
You can just mention in the commit message "and in one case, manually changing kwargs.keys() to kwargs" or something. I don't mind. :)